lang/java/maven-plugin/src/main/java/org/apache/avro/mojo/IDLProtocolMojo.java

```diff
@@ -80,16 +80,16 @@ protected void doCompile(String filename, File sourceDirectory, File outputDirec
         }
       }
 
-      URLClassLoader projPathLoader = new URLClassLoader(runtimeUrls.toArray(new URL[0]),
-          Thread.currentThread().getContextClassLoader());
-      try (Idl parser = new Idl(new File(sourceDirectory, filename), projPathLoader)) {
-
-        Protocol p = parser.CompilationUnit();
-        for (String warning : parser.getWarningsAfterParsing()) {
+      final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
Would it be a good idea to keep the old (JavaCC) code in a separate Mojo for a backup plan ?
I.e. if an application faces a bug in the new impl to be able to change the Maven plugin goal and use the JavaCC impl ?

Reply:
That's a bit tricky, as it requires a 2nd mojo, with e.g. 'old' in the name.
I've added a feature flag instead.